### PR TITLE
Fix and improve the hibernate reactive QuickStart following the Mutiny update

### PR DESCRIPTION
**Pull request by cescoffier**
CC @jponge

### CODE REVIEW

File: hibernate-reactive-quickstart/src/main/java/org/acme/hibernate/reactive/FruitMutinyResource.java [hunk 1]
```diff
@@ -88,8 +88,8 @@
     @Path("{id}")
     public Uni<Response> delete(@PathParam Integer id) {
         // Delete function (never returns null)
-        Function<Fruit, Uni<Response>> delete = entity -> mutinySession.remove(entity)
-                .chain(ignore -> mutinySession.flush())
+        Function<Fruit, Uni<? extends Response>> delete = entity -> mutinySession.remove(entity)
+                .chain(() -> mutinySession.flush())
```
Comment (jponge) [inline hibernate-reactive-quickstart/src/main/java/org/acme/hibernate/reactive/FruitMutinyResource.java:92]: `.chain(mutinySession::flush)`?

File: hibernate-reactive-routes-quickstart/src/main/java/org/acme/hibernate/reactive/FruitsRoutes.java [hunk 2]
```diff
@@ -80,7 +80,7 @@
         return session.find(Fruit.class, Integer.valueOf(id))
                 // If entity exists then delete
                 .onItem().ifNotNull().transformToUni(entity -> session.remove(entity)
-                        .chain(ignore -> session.flush())
+                        .chain(() -> session.flush())
```
Comment (jponge) [inline hibernate-reactive-routes-quickstart/src/main/java/org/acme/hibernate/reactive/FruitsRoutes.java:83]: Same here

**Comment by cescoffier** [comment]
All fixed.

**Comment by Sanne** [comment]
ah, thanks! Just did some duplicated work #664 . I'll close mine and merge this one.

**Comment by Sanne** [comment]
Would have been good to reply to https://github.com/quarkusio/quarkus/issues/6588#issuecomment-697147051 to avoid duplicating efforts

**Comment by cescoffier** [comment]
Didn't even see the notifications... weird.

**Pull request merged**